scripts/extrinsic_check.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+# This script:
+# - Checks for changes in transaction version in runtime/src/lib.rs
+# - Downloads latest release binary from gluwa/creditcoin (RELEASE_BIN)
+# - Compiles and build a binary from the current branch (HEAD_BIN)
+# - Runs the two nodes
+
+set -e
+
+HEAD_BIN=./target/release/creditcoin-node
+HEAD_WS=ws://localhost:9944
+RELEASE_WS=ws://localhost:9945
+
+runtimes=(
+  "creditcoin-node"
+)
+
+# First we fetch the latest released binary
+latest_release() {
+  curl -s "https://api.github.com/repos/$1/releases/latest" | jq -r '.tag_name'
+}
+
+latest_release=$(latest_release 'gluwa/creditcoin')
+RELEASE_BIN="./creditcoin-node"
+echo "[+] Fetching binary for Creditcoin version $latest_release"
+curl -L "https://github.com/gluwa/creditcoin/releases/download/$latest_release/creditcoin-v${latest_release}-x86_64-unknown-linux-gnu.tar.gz"  --output creditcoin.tar.gz && tar -xzf creditcoin.tar.gz || exit 1
+chmod +x "$RELEASE_BIN"
+git fetch --depth="${GIT_DEPTH:-100}" origin 'refs/tags/*:refs/tags/*'
+
+
+for RUNTIME in "${runtimes[@]}"; do
+  echo "[+] Checking runtime: ${RUNTIME}"
+
+  release_transaction_version=$(
+    git show "tags/$latest_release:runtime/src/lib.rs" | \
+      grep 'transaction_version'
+  )
+
+  current_transaction_version=$(
+    grep 'transaction_version' "./runtime/src/lib.rs"
+  )
+
+  echo "[+] Release: ${release_transaction_version}"
+  echo "[+] Ours: ${current_transaction_version}"
+
+  if [ ! "$release_transaction_version" = "$current_transaction_version" ]; then
+    echo "[+] Transaction version for ${RUNTIME} has been bumped since last release."
+    exit 0
+  fi
+
+  # Start running the nodes in the background
+  $HEAD_BIN --chain=local --tmp &
+  $RELEASE_BIN --chain=local --ws-port 9945 --tmp &
+  jobs
+
+  # Sleep a little to allow the nodes to spin up and start listening
+  TIMEOUT=30
+  for i in $(seq $TIMEOUT); do
+    sleep 1
+      if [ "$(lsof -nP -iTCP -sTCP:LISTEN | grep -c '994[45]')" == 2 ]; then
+        echo "[+] Both nodes listening"
+        break
+      fi
+      if [ "$i" == $TIMEOUT ]; then
+        echo "[!] Both nodes not listening after $i seconds. Exiting"
+        jobs -p | xargs kill; sleep 5
+        exit 1
+      fi
+  done
+  sleep 5
+
+  changed_extrinsics=$(
+    polkadot-js-metadata-cmp "$RELEASE_WS" "$HEAD_WS" \
+      | sed 's/^ \+//g' | grep -e 'idx: [0-9]\+ -> [0-9]\+' || true
+  )
+
+  if [ -n "$changed_extrinsics" ]; then
+    echo "[!] Extrinsics indexing/ordering has changed in the ${RUNTIME} runtime! If this change is intentional, please bump transaction_version in lib.rs. Changed extrinsics:"
+    echo "$changed_extrinsics"
+    exit 1
+  fi
+
+  echo "[+] No change in extrinsics ordering for the ${RUNTIME} runtime"
+  jobs -p | xargs kill; sleep 5
+done
+
+# Sleep a little to let the jobs die properly
+sleep 5
```

Comment:
Why not use `killall -9 creditcoin-node` on line 85 instead of waiting? It is not guaranteed that 5 seconds will be enough every time anyway.

Reply:
We are killing two processes; `creditcoin-node` and `./target/release/creditcoin-node`. Using process ID is more efficient.

Reply:
> We are killing two processes; creditcoin-node and ./target/release/creditcoin-node. Using process ID is more efficient.

This is exactly what `killall` does as well.

Reply:
I used `kill -9`.
`kill` takes PID as arg while `killall` takes process name.